x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/esnative/ReservedRealmElasticAutoconfigIntegTests.java

```diff
@@ -70,26 +64,7 @@ protected SecureString getBootstrapPassword() {
         return null; // no bootstrap password for this test
     }
 
-    private boolean isMigrationComplete(ClusterState state) {
-        IndexMetadata indexMetadata = state.metadata().getIndices().get(TestRestrictedIndices.INTERNAL_SECURITY_MAIN_INDEX_7);
-        return indexMetadata.getCustomData(MIGRATION_VERSION_CUSTOM_KEY) != null;
-    }
-
-    private void awaitSecurityMigrationRanOnce() {
```

Comment:
The `safeAwait(latch)` was failing consistently and this turned out not to be needed anymore now that security index is created on startup with migrations applied automatically.

Reply:
The `safeAwait` should not fail if this works as expected, even for a new index, the migration completed status should be there and the latch should complete. I think something is wrong if the `.security` index is created and doesn't have the migration status written at all (even if there is nothing to do, the latest migration version should still be written to the metadata). 

https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java#L1246-L1249

Reply:
There is some race condition here with the actual cause being:

```
[2025-01-24T09:00:37,924][WARN ][o.e.x.s.s.SecurityMigrationExecutor] [node_s_0] Security migration failed: org.elasticsearch.cluster.block.ClusterBlockException: index [.security-7] blocked by: [FORBIDDEN/13/cluster read-only / allow delete (api)];
```
I've reverted the change and moved the index deletion before the cluster is made read only.